schoenrock-rossiter_tom_ndvi.py

```diff
@@ -0,0 +1,656 @@
+study_sites = glob(os.path.join(site_path + "/*/"))
+print(study_sites)
+
+#function to open single landsat band, mask using qa and crop
+def open_clean_bands(band_path,
+                     crop_extent,
+                     valid_range=None,):
+    """Open and mask a single landsat band using a pixel_qa layer.
+
+    Parameters
+    -----------
+    band_path : string
+        A path to the array to be opened
+    crop_extent : vector
+        A path to the vector file to use as crop boundary
+    valid_range : tuple (optional)
+        A tuple of min and max range of values for the data. Default = None
+
+
+    Returns
+    -----------
+    arr : xarray DataArray
+        An xarray DataArray with values that should be masked set to 1 for 
+        True (Boolean)
+    """
+    band = rxr.open_rasterio(band_path, masked = True).rio.clip    (crop_extent.geometry, from_disk = True).squeeze()
+
+    # Only run this step if a valid range tuple is provided
+    if valid_range:
+        mask = ((band < valid_range[0]) | (band > valid_range[1]))
+        band = band.where(~xr.where(mask, True, False))
+
+    return band
+
+# function to apply cloud mask and caluclate ndvi (mean)
+def mask_crop_ndvi(all_bands,
+                   crop_bound,
+                   pixel_qa,
+                   masked_values):
+    """Open and mask a single landsat band using a pixel_qa layer.
+
+    Parameters
+    -----------
+    all_bands : list
+        a list containing the xarray objects for landsat  bands 4 and  5
+    crop_bound: geopandas GeoDataFrame
+        A geopandas dataframe to be used to crop the raster data using
+        rasterio mask().
+    pixel_qa: xarray DataArray
+        An xarray DataArray with pixel qa values that have not yet been
+        turned into a mask (0s and 1s)
+    masked_values: list
+        A list of cloud mask values
+
+
+    Returns
+    -----------
+    ndvi_crop : Xarray Dataset
+        a cropped and masked xarray object containing NDVI values
+    """
+
+    crop_json = crop_bound.geometry
+
+    # Clip pixel qa cloud mask layer
+    cl_mask_crop = pixel_qa.rio.clip(crop_json)
+
+    # Calculate NDVI
+    ndvi_xr = (all_bands[1]-all_bands[0]) / (all_bands[1]+all_bands[0])
+    # Clip NDVI layer
+    ndvi_crop = ndvi_xr.rio.clip(crop_json)
+
+    # Apply cloud mask to NDVI
+    ndvi_crop = ndvi_crop.where(~pixel_qa.isin(masked_values))
+
+    
+    return ndvi_crop.mean().values
+    
+#third function containing the above two
+def output_ndvi(scene_path, crop_extent):
+    """Function that combines the above two functions to open, stack, crop
+    and mask Landsat scenes.
+
+    Parameters
+    -----------
+    scene_path : string
+        A path to the array to be opened
+    crop_extent: vector
+        A path to the vector file to use as crop boundary
+    
+    Returns
+    -----------
+    ndvi_crop : ndarray Dataset
+        a cropped and masked ndarray object containing mean NDVI values
+    """
+    
```

Comment:
Comments would be helpful to explain what are we doing in the three lines of code below.